src/main/java/UseCase/CreatePlaylistInteractor.java

```diff
@@ -0,0 +1,51 @@
+package UseCase;
+import Database.playlistDsData;
+import Entities.Playlist;
+import InputBoundary.newPlaylistInputBoundary;
+import InputData.newPlaylistInputData;
+import Database.playlistAccessInterface;
+import Database.playlistLibrary;
+import OutputBoundary.newPlaylistOutputBoundary;
+import OutputData.newPlaylistOutputData;
+
+public class CreatePlaylistInteractor implements newPlaylistInputBoundary {
+        private final newPlaylistOutputBoundary presenter;
+
+        private final playlistAccessInterface library;
+    /**
+     *
+     * @param presenter presenter object to store output message in
+     */
+    public CreatePlaylistInteractor(newPlaylistOutputBoundary presenter){
+        this.presenter = presenter;
+        this.library = playlistLibrary.getInstance();
+    }
+    /**
+     * @param newplaylistInputData holds necessary data to instantiate a new playlist
+     * if hasFirstSong() is false, then newPlaylist() will instantiate an empty playlist and vice versa if it's value
+     * is true
+     */
+    public void newPlaylist(newPlaylistInputData newplaylistInputData) {
+        Playlist playlist;
+        if (!newplaylistInputData.hasFirstSong()) {
+            playlist = new Playlist(newplaylistInputData.getId(), newplaylistInputData.getPlaylistName(),
+                    newplaylistInputData.getOwner());
+        }
+        else{
+            playlist = new Playlist(newplaylistInputData.getId(), newplaylistInputData.getPlaylistName(), newplaylistInputData.getOwner(),
+                     newplaylistInputData.getFirstSong());
+        }
+        newplaylistInputData.getOwner().addPlaylist(playlist);
```

Comment:
This is a great implementation and correct use of the playlist database. However, I am wondering if you will also have to use the user database class to save the user, after assigning them the new playlist.

I think you will have to do this, since there is no connection between the RegularUser class and the user database to say, "Hey, I have a new playlist that needs to be added to my data and saved!". So, I think you'll have to add the instance variable `userAccessInterface userDatabase` to the top of your file, and then instantiate it similarly to the playlist library, and then call the `userDatabase.save()` method on the `RegularUser` after you add the playlist to it in line 38. This will ensure that the new playlist is correctly written to the database as belonging to the user.